OpenRA.Mods.D2k/Traits/Spicebloom.cs

```diff
@@ -0,0 +1,194 @@
+	{
+		public readonly string[] GrowthSequences = { "grow1", "grow2", "grow3" };
+
+		[Desc("The amount of time (in ticks) that the spicebloom will take to respawn.")]
+		public readonly int[] RespawnTime = { 750, 1500 };
+
+		[Desc("The amount of time (in ticks) that the spicebloom will take to grow.")]
+		public readonly int[] GrowthTime = { 750, 1500 };
+
+		public readonly string ResourceType = "Spice";
+
+		[Desc("The weapon to use as spice")]
+		[WeaponReference]
+		public readonly string Weapon = "SpiceExplosion";
+
+		[Desc("The amount of spice to expel.")]
+		public readonly int Pieces = 7;
+
+		[Desc("The maximum distance in cells that spice may be expelled.")]
+		public readonly int Range = 5;
+
+		public object Create(ActorInitializer init)
+		{
+			return new Spicebloom(init, this);
+		}
+	}
+
+	public class Spicebloom : ITick, ICrushable, INotifyDamage
+	{
+		readonly Actor self;
+		readonly SpicebloomInfo info;
+		readonly ResourceType resType;
+		readonly ResourceLayer resLayer;
+		readonly RenderSprites render;
+
+		Animation anim;
+		AnimationWithOffset anim2;
+		int explodedTicks;
+		bool growing;
+		int growTicks;
+		[Sync] int respawnTime;
+		[Sync] int growTime;
+
+		public Spicebloom(ActorInitializer init, SpicebloomInfo info)
+		{
+			this.info = info;
+			self = init.Self;
+
+			resType = self.World.WorldActor.TraitsImplementing<ResourceType>()
+							.FirstOrDefault(t => t.Info.Name == info.ResourceType);
+
+			if (resType == null)
+				throw new InvalidOperationException("No such resource type `{0}`".F(info.ResourceType));
+
+			resLayer = self.World.WorldActor.Trait<ResourceLayer>();
+
+			render = self.Trait<RenderSprites>();
+
+			anim = new Animation(init.Self.World, render.GetImage(self));
+			anim2 = new AnimationWithOffset(anim, null, () => false);
+			Respawn();
+		}
+
+		public void Tick(Actor self)
+		{
+			if (growing)
+			{
+				growTicks++;
+
+				if (growTicks >= growTime)
+					Explode();
+				else
+				{
+					var index = info.GrowthSequences.Length * (growTicks / growTime);
+					anim.Play(RenderSprites.NormalizeSequence(anim, self.GetDamageState(), info.GrowthSequences[index]));
+				}
+			}
+			else
+			{
+				explodedTicks++;
+
+				if (explodedTicks >= respawnTime)
+					Respawn();
+			}
+		}
+
+		public void OnCrush(Actor crusher)
+		{
+			if(growing)
+				Explode();
+		}
+
+		public void WarnCrush(Actor crusher) { }
+
+		public bool CrushableBy(string[] crushClasses, Player owner) 
+		{
+			return true;
+		}
+
+		public void Damaged(Actor self, AttackInfo e)
+		{
+			if (growing)
+				Explode();
+		}
+
+		void Respawn()
+		{
+			render.Add(anim2);
+
+			explodedTicks = 0;
+			growing = true;
+			growTicks = 0;
+			respawnTime = self.World.SharedRandom.Next(info.RespawnTime[0], info.RespawnTime[1]);
+			growTime = self.World.SharedRandom.Next(info.GrowthTime[0], info.GrowthTime[1]);
+		}
+
+		void Explode()
+		{
+			//render.Remove(anim2);
```

Comment:
> [SA1512] A single-line comment must not be followed by a blank line. To ignore this error when commenting out a line of code, begin the comment with '////' rather than '//'.